src/problem.jl

```diff
@@ -122,9 +122,12 @@ end
 # when setindex! is used.
 function Base.setindex!(prob::JumpProblem, args...; kwargs...)
     SciMLBase.___internal_setindex!(prob.prob, args...; kwargs...)
-    if using_params(prob.massaction_jump)
-        update_parameters!(prob.massaction_jump, prob.prob.p)
-    end
+
+    # since parameters are no longer allowed to be mutated and the preceding will error this
+    # isn't needed
+    # if using_params(prob.massaction_jump)
+    #     update_parameters!(prob.massaction_jump, prob.prob.p)
+    # end
```

Comment:
`setp` will still allow for issues, along with manual mutation.

Reply:
We probably need a custom dispatch on it then to update MassActionJumps?

Reply:
Yes, @AayushSabharwal can probably help with that.

Reply:
That would be great, thanks!

Reply:
We now have `set_parameter!` which is used by `setp` internally, and can be implemented to do other bookkeeping actions when `setp` is used.

Reply:
@AayushSabharwal, can you add the needed dispatches to that when parameters get modified, i.e. something like what we had before when modifying a problem:

```julia
if using_params(prob.massaction_jump)
    update_parameters!(prob.massaction_jump, prob.prob.p)
end
```

Alternatively, @ChrisRackauckas do we want to put this on users (since this could be expensive, so a faster workflow is to only call it once all parameter modifications are finished)? That does leave a gotcha someone could encounter unless they carefully read the docs (which is why the previous behavior here always updated directly after modifying a parameter).

Reply:
But we do put calling the jump update function on users currently when modifying integrators within callbacks, so perhaps it isn't unreasonable to do that.